lib/java/com/google/android/material/button/MaterialButton.java

```diff
@@ -157,8 +157,22 @@
    */
   public static final int ICON_GRAVITY_TEXT_END = 0x4;
 
+  /**
+   * Gravity used to position the icon in the center of the view on top of the text
+   *
+   * @see #setIconGravity(int)
+   * @see #getIconGravity()
+   */
+  public static final int ICON_GRAVITY_TOP = 0x5;
```

Comment:
Can you please also add support for ICON_GRAVITY_TEXT_TOP?

Reply:
Can you please explain the difference between ICON_GRAVITY_TOP  and ICON_GRAVITY_TEXT_TOP?

If the idea is to keep the icon closer to the top of the view (ICON_GRAVITY_TOP) or closer to the top of the text (ICON_GRAVITY_TEXT_TOP), in my point of view, both can be achieved by adjusting iconPadding and paddingTop.

Reply:
IMO adjusting `iconPadding` along with `paddingTop` is hacky and not flexible enough.

Reply:
Thanks for the suggestion, after adding the support to textTop and playing around with the feature I've noticed that it's really handy to have it. 👍 

Solution add on d43aa24